### PR TITLE
Fix iOS Begin button crash from invalid currentDay (#250)

### LINKED ISSUE #250: iOS: app crashes when tapping Begin button (regression in build 8)
## Summary

Tapping the **Begin** button on the iOS Home screen crashes the app immediately — the timer screen never loads. This is a hard regression introduced between the previously-shipped TestFlight/App Store build and the new build 8 that just released ([PR #245](https://github.com/auerbachb/still-point/pull/245)).

## Repro

1. Install the latest TestFlight / App Store build (`1.0.3 (8)`) on a physical iPhone
2. Sign in
3. From the Home screen, tap **Begin**
4. App crashes before the timer view appears

## Expected

Tapping Begin transitions to `SessionView` and starts the countdown timer.

## Actual

App crashes synchronously on the navigation transition / `SessionView` mount, before any timer UI renders.

## Suspect surface

The Begin button calls `appVM.beginSession()` which only flips `currentView = .session` ([AppViewModel.swift:115](ios/StillPointApp/ViewModels/AppViewModel.swift#L115)). The crash therefore happens during `SessionView` mount or `SessionViewModel.start()` — likely candidates:

- [ios/StillPointApp/Views/SessionView.swift](ios/StillPointApp/Views/SessionView.swift) — `init` constructs `SessionViewModel`; `onAppear` calls `vm.start()`
- [ios/StillPointApp/ViewModels/SessionViewModel.swift](ios/StillPointApp/ViewModels/SessionViewModel.swift) — `init` reads `AudioEngine.loadPrefs()`; `start()` calls `AudioEngine.shared.warmUp()` and starts a `Timer.publish` Combine timer
- [ios/StillPointShared/Sources/StillPointShared/AudioEngine.swift](ios/StillPointShared/Sources/StillPointShared/AudioEngine.swift) — audio session activation

PR #245 itself only changed `AuthViewModel` + project metadata. The actual breaking change must have landed earlier and was first delivered to production in build 8. Use `git log ios/ --since=<prior-shipped-build-date>` to bisect.

## Investigation plan

1. **Get a crash log.** Pull the most recent crash from the device (Settings → Privacy → Analytics & Improvements → Analytics Data, look for `StillPointApp-*.ips`) or Xcode Organizer → Crashes. Symbolicate it. The stack trace will pinpoint the offending call.
2. **Reproduce locally** by archiving the same configuration (Release, signed) and running on device. The crash may not reproduce in Debug.
3. **Bisect commits** between the prior shipped build's commit and `3757048` (PR #245 merge) limited to `ios/`.
4. **Fix the root cause** — likely a force-unwrap, a missing audio-session entitlement, a SwiftData schema mismatch, or a Release-only optimization issue.
5. **Add a regression test** — see the companion issue on E2E coverage that should have caught this pre-release.

## Acceptance criteria

- [ ] Crash log obtained and root cause identified
- [ ] Fix landed; tapping Begin reliably loads the timer on a Release-built device install
- [ ] Smoke-tested on a physical device against the Release configuration (not just simulator Debug)
- [ ] New build cut and verified on TestFlight before any further App Store submission
- [ ] A regression test exists that would have failed against build 8 (coordinated with the E2E issue)

## Severity

P0 — currently-shipped App Store build is non-functional for new sessions.

---

## Implementation Plan (merged: Claude's investigation + CodeRabbit's root-cause hypothesis)

> Combines Claude's broader investigation/regression framing with CodeRabbit's specific root-cause hypothesis. CR's hypothesis is a plausible, targeted fix; Claude's investigation is still required because CR's hypothesis has a logical gap that needs verification.

### CodeRabbit's root-cause hypothesis

[`StillPoint.duration(forDay:)`](ios/StillPointShared/Sources/StillPointShared/Constants.swift#L14-L15) has a hard `precondition(day >= 1, "Day must be >= 1")`. [`AppViewModel.currentDay`](ios/StillPointApp/ViewModels/AppViewModel.swift#L51-L53) returns `currentUser?.currentDay ?? 1` — the `?? 1` only handles `nil`, **not** `0` or negative values. If the server returns `currentDay: 0` (or the user has just registered and a day-counter bug exists), the precondition fires and the app crashes.

CR proposes a one-line defensive clamp: change `currentUser?.currentDay ?? 1` to `max(currentUser?.currentDay ?? 1, 1)`.

### Claude's gap-check on CR's hypothesis

[`HomeView.swift:36`](ios/StillPointApp/Views/HomeView.swift#L36) renders `appVM.todayDuration`, which calls the same `StillPoint.duration(forDay: currentDay)` precondition path. If `currentDay` were `< 1`, **Home itself would crash before the user could tap Begin** — but the user reports seeing Home and successfully tapping Begin. This means either:

1. `currentDay` is `>= 1` at Home render time but somehow becomes `< 1` between Home render and Session mount (unlikely without an interleaving network refresh)
2. The crash is on a different code path (e.g., `SessionViewModel.init`'s call to `AudioEngine.loadPrefs()`, the `Timer.publish` Combine pipeline, audio-session activation in `AudioEngine.shared.warmUp()`, or a Release-only optimization issue)
3. CR's hypothesis is correct but only manifests on Begin tap because HomeView's `todayDuration` is wrapped by SwiftUI in a way that swallows the precondition crash on first render, and SessionView's synchronous `init` call is the first one that actually traps

A real crash log resolves this in seconds — without it we are guessing.

### Combined plan (do both, in order)

1. **Get the crash log first.** Pull the `.ips` from the device (Settings → Privacy & Security → Analytics & Improvements → Analytics Data, look for `StillPointApp-*.ips`) or open Xcode → Window → Organizer → Crashes. Symbolicate. The stack trace is the source of truth — do not guess past this step.
2. **Apply CR's defensive clamp regardless of what the crash log says.** It is a strict improvement: `currentUser?.currentDay ?? 1` → `max(currentUser?.currentDay ?? 1, 1)` in [`AppViewModel.swift:51-53`](ios/StillPointApp/ViewModels/AppViewModel.swift#L51-L53). Even if it isn't the proximate cause of *this* crash, returning `0`/negative from the server is a latent landmine that should be guarded.
3. **Audit other call sites of `StillPoint.duration(forDay:)` and `StillPoint.blockCount(forDuration:)`** for the same pattern. Either clamp at every call site, or — better — **move the clamp into `Constants.swift`**: replace `precondition(day >= 1)` with `let day = max(day, 1)` (or document why preconditions are intentionally fatal in production and clamp at all callers). Pick one approach and apply it consistently.
4. **Bisect** if the crash log shows a different culprit. `git log ios/ --since=<prior-shipped-build-date>` between the previously-shipped build's commit and `3757048` (PR #245 merge). Likely candidates that landed in this range:
   - PR #228 (note save reliability) — touches `BatchThoughtsRequest` paths; unlikely to crash on Begin
   - PR #234 (password reset flow) — touches Auth; unlikely to crash on Begin
   - The shared package may have audio/SwiftData changes worth re-reading
5. **Server-side check.** Pull this user's `currentDay` from the database. If it's `0` or negative, fix the data and find the writer that produced it (registration flow? day-rollover cron?).
6. **Add regression coverage.** Tests should fail against the build-8 binary (or a faithful reproduction):
   - **Unit:** `AppViewModel.currentDay` returns `1` for nil, `1` for `0`, `1` for negative, and the actual value for `>= 1`.
   - **Unit:** `StillPoint.duration(forDay:)` either clamps or `precondition`s — pin the chosen contract with a test.
   - **UI/E2E:** A test that taps Begin and asserts `session.timerLabel` exists. This already exists at [StillPointAppUITests.swift:63-68](ios/StillPointAppUITests/StillPointAppUITests.swift#L63-L68); the meta-question of why it didn't catch this lives in #253.
7. **Verify on a Release-signed device build before tagging.** A simulator-Debug pass is not enough — that's why this shipped. See #253 for the durable fix.

### Acceptance criteria (revised, supersedes the original)

- [ ] Crash log obtained and root cause confirmed
- [ ] `AppViewModel.currentDay` clamped to `>= 1` (CR's fix)
- [ ] All call sites of `StillPoint.duration(forDay:)` audited; clamping/precondition strategy applied consistently
- [ ] Unit test added for `currentDay` edge cases (nil / 0 / negative / valid)
- [ ] If a different root cause was identified in the crash log, it is also fixed
- [ ] If server returned bad data, the source is identified and tracked (separate issue if needed)
- [ ] Release-signed device install verified: tap Begin, full session completes, returns to Home
- [ ] New build cut and verified on TestFlight before App Store submission
- [ ] Regression test exists that would fail against build 8's binary

### PR DESCRIPTION
**Pull request by auerbachb**
## Summary

Tapping **Begin** on the iOS Home screen was crashing the app on the App Store build (1.0.3 build 8) before the timer view could render — a hard regression reported in #250.

CodeRabbit's plan-on-issue identified the proximate cause: [`StillPoint.duration(forDay:)`](ios/StillPointShared/Sources/StillPointShared/Constants.swift#L14-L17) has a `precondition(day >= 1)` that traps when the server returns `currentDay: 0` (or negative), and [`AppViewModel.currentDay`](ios/StillPointApp/ViewModels/AppViewModel.swift#L51-L55) only guarded `nil` with `?? 1`. This PR applies CR's targeted clamp **and** adds defense-in-depth at the precondition site so production cannot crash from bad upstream day numbers regardless of caller.

## Changes

- **[AppViewModel.swift](ios/StillPointApp/ViewModels/AppViewModel.swift#L51-L55):** `currentDay` now returns `max(currentUser?.currentDay ?? 1, 1)`. Protects the `SessionView` mount path that the Begin button triggers.
- **[Constants.swift](ios/StillPointShared/Sources/StillPointShared/Constants.swift#L13-L21):** `StillPoint.duration(forDay:)` replaces the fatal `precondition` with a debug-only `assert` + `max(day, 1)` clamp. Dev still gets the trap signal for upstream bugs; Release builds survive bad data.
- **[StillPointDurationTests.swift](ios/StillPointShared/Tests/StillPointSharedTests/StillPointDurationTests.swift):** new tests pinning the duration formula at `day == 1` and beyond, plus the `blockCount` rounding behavior.

## Why both fixes

CR's clamp at `AppViewModel.currentDay` is correct and minimal. But the same precondition is reachable through other call sites ([`SessionViewModel.swift:340`](ios/StillPointApp/ViewModels/SessionViewModel.swift#L340), [`CompletionView.swift:19`](ios/StillPointApp/Views/CompletionView.swift#L19)), and [`HomeView.swift:36`](ios/StillPointApp/Views/HomeView.swift#L36) already exercises that path before Begin is tapped — so a future bug producing a bad day number anywhere should not be able to crash users in production. Hardening the function itself is a strict win and consistent with the existing `max(store.user.currentDay, 1)` clamps already present in [`APIClient.swift:374`](ios/StillPointShared/Sources/StillPointShared/APIClient.swift#L374) and [`:475`](ios/StillPointShared/Sources/StillPointShared/APIClient.swift#L475).

## What this PR does **not** cover

- **Crash log not yet obtained** — without the symbolicated `.ips` we can't 100% confirm this was the proximate cause vs. some other Release-only path (e.g. `AudioEngine` warmUp). The fix is defense-in-depth either way; if a different root cause exists it's a separate fix on top.
- **Server-side data audit** — if real users ended up with `currentDay: 0` in the database, the writer that produced that needs investigation. Tracked separately.
- **Why the existing E2E test didn't catch this** — covered in #253.
- **End-of-session note save error** — covered in #254.

## Test plan

- [ ] `swift test` from `ios/StillPointShared/` passes (new + existing tests)
- [ ] iOS app builds cleanly under Release configuration (Xcode archive)
- [ ] On a Release-signed device install: tapping Begin loads the timer (no crash)
- [ ] On a Release-signed device install: completing a session reaches CompletionView
- [ ] CR review on the PR is clean

Closes #250

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Prevents crashes from invalid "day" values by clamping inputs to a safe minimum and using debug-only checks.
  * Ensures the app reports a safe current day when user data is missing or out-of-range, fixing downstream duration/display issues.

* **Tests**
  * Added tests for duration, block-count calculations, and day-clamping/validation behavior.

* **Chores**
  * Ignore local agent runtime lockfile to avoid committing it.

### CODE REVIEW

File: ios/StillPointApp/ViewModels/AppViewModel.swift [hunk 1]
```diff
@@ -49,7 +49,9 @@ final class AppViewModel {
     private var pendingBuddyInviteToken: String?
 
     var currentDay: Int {
-        currentUser?.currentDay ?? 1
+        // Server may transiently return 0/negative; clamp to keep `StillPoint.duration(forDay:)`
+        // out of its defensive trap and avoid crashing the session UI.
+        max(currentUser?.currentDay ?? 1, 1)
     }
```
[the remark this answers is not in the document]
Comment (auerbachb) [reply ios/StillPointApp/ViewModels/AppViewModel.swift]: Fixed in `4b23e2f`: extracted the clamp into a testable shared helper `StillPoint.clampedCurrentDay(for:)` and added 4 edge-case tests in `StillPointDurationTests.swift` covering nil / 0 / negative / valid positive inputs. The iOS app target has no XCTest unit-test target, so co-locating the testable surface in the shared package was the cleanest path.

File: ios/StillPointShared/Sources/StillPointShared/Constants.swift [hunk 2]
```diff
@@ -10,14 +10,28 @@ public enum StillPoint {
     /// Visual block duration in seconds
     public static let blockDuration = 10
 
-    /// Calculate session duration for a given day number
+    /// Calculate session duration for a given day number.
+    ///
+    /// Clamps `day` to `>= 1` so production code never traps on bad upstream data
+    /// (e.g. a server returning `0`). A debug-only `assertionFailure` still surfaces
+    /// the upstream bug during development.
```
[the remark this answers is not in the document]
Comment (auerbachb) [reply ios/StillPointShared/Sources/StillPointShared/Constants.swift]: Fixed in `11bbd1b`: doc comment now says `assert` to match the implementation.

**Comment by auerbachb** [comment]
@coderabbitai full review.

**Comment by auerbachb** [comment]
@cursor review
